tests/test_continuous_futures.py

```diff
@@ -1523,6 +1523,35 @@ def test_end_before_auto_close(self):
             ],
         )
 
+    def test_get_contract_center(self):
+        asset_finder = self.asset_finder
+        get_contract_center = partial(
+            self.volume_roll_finder.get_contract_center,
+            root_symbol='CL',
+            offset=0,
+        )
+
+        # Test that the current contract adheres to the rolls.
+        self.assertEqual(
+            get_contract_center(dt=pd.Timestamp('2017-01-18', tz='UTC')),
+            self.asset_finder.retrieve_asset(1000),
+        )
+        self.assertEqual(
+            get_contract_center(dt=pd.Timestamp('2017-01-19', tz='UTC')),
+            self.asset_finder.retrieve_asset(1001),
+        )
+
+        # Test that we still get the correct current contract close to the max
+        # day boundary.
+        self.assertEqual(
+            get_contract_center(dt=pd.Timestamp('2017-04-17', tz='UTC')),
+            self.asset_finder.retrieve_asset(1003),
+        )
+        self.assertEqual(
+            get_contract_center(dt=pd.Timestamp('2017-04-18', tz='UTC')),
```

Comment:
I'm thinking it might be more maintainable to define these dates relative to `self.END_DATE`, i.e. in https://github.com/quantopian/zipline/pull/1917 I pushed the end date back, which would have broken the intention of these tests and I don't think I would have realized it.